File: examples/python/export_events.py
"""Export audit events for compliance reporting."""

import requests

API_KEY = "stl_live_..."
BASE_URL = "https://api.sealtrail.dev"
HEADERS = {"Authorization": f"Bearer {API_KEY}"}

# Export as CSV
csv_res = requests.get(
    f"{BASE_URL}/v1/events/export",
    headers=HEADERS,
    params={
        "format": "csv",
        "after": "2026-01-01T00:00:00Z",
        "before": "2026-03-18T00:00:00Z",
    },
)
csv_res.raise_for_status()

with open("audit-export.csv", "w") as f:
    f.write(csv_res.text)
print(f"CSV export saved ({len(csv_res.text)} bytes)")

# Export as JSON with filters
json_res = requests.get(
    f"{BASE_URL}/v1/events/export",
    headers=HEADERS,
    params={
        "format": "json",
        "after": "2026-01-01T00:00:00Z",
        "before": "2026-03-18T00:00:00Z",
        "actor": "user_123",
    },
)
json_res.raise_for_status()

data = json_res.json()
print(f"{data['export']['count']} events exported")
print(f"Period: {data['export']['period']['after']} to {data['export']['period']['before']}")

for event in data["data"]:
    print(f"  {event['timestamp']} | {event['actor']} | {event['action']}")

File: examples/typescript/export-events.ts
// Export audit events as CSV (raw fetch, no SDK)
const params = new URLSearchParams({
  format: "csv",
  after: "2026-01-01T00:00:00Z",
  before: "2026-03-18T00:00:00Z",
});

const csvRes = await fetch(
  `https://api.sealtrail.dev/v1/events/export?${params}`,
  { headers: { Authorization: "Bearer stl_live_..." } }
);

const csv = await csvRes.text();
console.log("CSV preview (first 3 lines):");
console.log(csv.split("\n").slice(0, 3).join("\n"));

// Export as JSON with filters
const jsonParams = new URLSearchParams({
  format: "json",
  after: "2026-01-01T00:00:00Z",
  before: "2026-03-18T00:00:00Z",
  actor: "user_123",
});

const jsonRes = await fetch(
  `https://api.sealtrail.dev/v1/events/export?${jsonParams}`,
  { headers: { Authorization: "Bearer stl_live_..." } }
);

const { export: meta, data } = await jsonRes.json();
console.log(`${meta.count} events exported`);
